jbehave-support-core/src/main/java/org/jbehavesupport/core/web/WebSteps.java

```diff
@@ -190,6 +198,8 @@ public void waitUntilCondition(String page, String element, ExpressionEvaluating
         WebWaitCondition waitCondition = waitConditionResolver.resolveWaitCondition(waitConditionCtx);
 
         waitCondition.evaluate(waitConditionCtx);
+        screenshotCreator.createScreenshot(WebScreenshotCreator.type.WAIT);
```

Comment:
Do we need to call both? Wouldnt it be easier to have the WAIT type included in the STEP type?
Hierarchy like MANUAL < WAIT < STEP < DEBUG.. Or would that complicate things even further without much benefits?

Reply:
It would make if in CreateScreenshot really ugly.
`if(screenShotType == Type.FAILED || 
   screenShotType == Type.MANUAL || 
   screenShotType == desiredType || 
   (screenShotType == Type.WAIT && desiredType == Type.STEP)) ||
   ((screenShotType == Type.STEP || screenShotType == Type.WAIT) && desiredType == Type.DEBUG)){...}`
But I can do it, what do you think?

Reply:
Hmm ok, that looks even more complicated... How about something like enums with internal value? E.g. https://stackoverflow.com/a/44654387? (Or using ordinal method, but I dont think its a good idea in this case)
But just suggesting alternative, dont know if it would make the code even more complicated or not.. Just seems like something that could be done in _some_ simpler way 🙂

Reply:
Thanks for suggestion!